subsys/net/ip/ipv4.c

```diff
@@ -139,6 +139,10 @@ int net_ipv4_parse_hdr_options(struct net_pkt *pkt,
 			return -EINVAL;
 		}
 
+		if (opts_len < 1U) {
```

Comment:
As we need to have a new version, it would be a good idea to rename the opts_len to total_opt_len or similar in order it to not get mixed up with opt_len variable.

Reply:
Done.